### PR TITLE
[Internal] Thin Client Integration: Adds proxy request headers startEPK and endEPK for query operation.

### LINKED ISSUE #5185: [Thin Client Integration] Bad request error for query operation due to missing headers.
**Describe the bug**
Bad request error for query operation due to missing headers. Need to add startEPK and endEPK to proxy request for query operation.

### PR DESCRIPTION
**Pull request by aavasthy**
# Pull Request Template

## Description

Adds proxy request headers  startEPK and endEPK for query operation in thinclient mode.
Also, updated and cleaned up ThinClientStoreClient unit tests.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

## Closing issues

To automatically close an issue: closes #5185

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/ThinClientStoreClient.cs [hunk 1]
```diff
@@ -90,7 +90,17 @@ private async ValueTask<HttpRequestMessage> PrepareRequestForProxyAsync(
 
             BufferProviderWrapper bufferProviderWrapper = this.bufferProviderWrapperPool.Get();
             try
-            {
+            {
+                PartitionKeyRange partitionKeyRange = request.RequestContext.ResolvedPartitionKeyRange;
+
+                requestMessage.Headers.TryAddWithoutValidation(
```
Comment (kundadebdatta) [inline Microsoft.Azure.Cosmos/src/ThinClientStoreClient.cs:96]: This would mean that even if `partitionKeyRange` is `null`, we will end up adding the null value in the header correct ? Howe about having the null check before adding it to the headers ?

Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/ThinClientStoreClient.cs:96]: Ut coverage?

Comment (aavasthy) [reply Microsoft.Azure.Cosmos/src/ThinClientStoreClient.cs:96]: In serializer we do check for the null for the same header, but I can include it here.

Comment (FabianMeiswinkel) [reply Microsoft.Azure.Cosmos/src/ThinClientStoreClient.cs:96]: Please add a unit test for PK==Null and PK==None

**Review by kundadebdatta — APPROVED** [review]
LGTM. The PK `Null` and `None` tests will be added as a part of a follow up PR correct ?

File: Microsoft.Azure.Cosmos/src/ThinClientStoreClient.cs [hunk 2]
```diff
@@ -90,7 +90,20 @@ private async ValueTask<HttpRequestMessage> PrepareRequestForProxyAsync(
 
             BufferProviderWrapper bufferProviderWrapper = this.bufferProviderWrapperPool.Get();
             try
-            {
+            {
+                PartitionKeyRange partitionKeyRange = request.RequestContext.ResolvedPartitionKeyRange;
```
Comment (FabianMeiswinkel) [inline Microsoft.Azure.Cosmos/src/ThinClientStoreClient.cs:94]: This looks weird - is there any guarantee that ResolvedPartitionKeyRange will only ever be set when only routing information is PhysicalPartitionKeyRangeId?

Wouldn't this property also be set when using ParittionKeyValue or EPK for routing? In which case the startEpk/EndEpk should be the narrower scope?

Comment (FabianMeiswinkel) [reply Microsoft.Azure.Cosmos/src/ThinClientStoreClient.cs:94]: HOnes question - what lead to choosing existence of ResolvedPartitionKeyrange to set these headers?

Comment (aavasthy) [reply Microsoft.Azure.Cosmos/src/ThinClientStoreClient.cs:94]: This  startEpk/EndEpk headers are only being used for non-point operations. That check is added in the thinclient serializer code https://github.com/Azure/azure-cosmos-dotnet-v3/blob/ea26d298279242e019b1317a498fe53b44d0511c/Microsoft.Azure.Cosmos/src/ThinClientTransportSerializer.cs#L91

Comment (FabianMeiswinkel) [reply Microsoft.Azure.Cosmos/src/ThinClientStoreClient.cs]: Sounds good

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM now - thanks!

**Review by kundadebdatta — APPROVED** [review]
LGTM.

File: Microsoft.Azure.Cosmos/src/ThinClientTransportSerializer.cs [hunk 3]
```diff
@@ -74,11 +74,11 @@ public static async Task<Stream> SerializeProxyRequestAsync(
 
             if (operationType.IsPointOperation())
             {
-                string partitionKey = request.Headers.Get(HttpConstants.HttpHeaders.PartitionKey);
-                
-                if (string.IsNullOrEmpty(partitionKey))
-                {
-                    throw new InternalServerErrorException();
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/ThinClientTransportSerializer.cs:81]: nit: Revert

**Review by kirankumarkolli — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/ThinClientTransportSerializer.cs (hunk 3 above)
Comment (aavasthy) [reply Microsoft.Azure.Cosmos/src/ThinClientTransportSerializer.cs:81]: will clean it up as part of follow up with more test changes https://github.com/Azure/azure-cosmos-dotnet-v3/pull/5214
